File: Cc.Ui/src/models/tabCountry.ts
import { Entity, resource, field, association } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('tabCountry')
export class TabCountry extends Entity {

    @field()
    id: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;    
    @field()
    code: string;
    @field()
    name: string;
    @field()
    localName?: string;
    @field()
    citizenship?: string;
    @field()
    iso2Code?: string;
    
    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): TabCountry {

        let item = new TabCountry();
        return item;
    }
}

@resource('tabCountry')
export class TabCountrySearchModel extends Entity implements Pagination {
        
    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}

File: Cc.Core/Models/TabCountry.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class TabCountry
    {
        public long Id { get; set; }
        public DateTime InsertDate { get; set; }
        public string? InsertUser { get; set; }
        public DateTime UpdateDate { get; set; }
        public string? UpdateUser { get; set; }
        public string Code { get; set; } = null!;
        public string Name { get; set; } = null!;
        public string? LocalName { get; set; }
        public string? Citizenship { get; set; }
        public string? Iso2Code { get; set; }
        public string? Iso3Code { get; set; }
    }
}
